ahnlich/ai/src/engine/store.rs

```diff
@@ -100,28 +106,57 @@ impl AIStoreHandler {
     }
 
     /// Converts storeinput into a tuple of storekey and storevalue.
-    /// Fails if the type of storeinput does not match the store type
+    /// Fails if the store input type does not match the store index_type
     #[tracing::instrument(skip(self))]
     pub(crate) fn store_input_to_store_key_val(
         &self,
         store_name: &StoreName,
         store_input: StoreInput,
         store_value: &StoreValue,
+        preprocess_action: &PreprocessAction,
     ) -> Result<(StoreKey, StoreValue), AIProxyError> {
         let metadata_key = &*AHNLICH_AI_RESERVED_META_KEY;
         if store_value.contains_key(metadata_key) {
             return Err(AIProxyError::ReservedError(metadata_key.to_string()));
         }
         let store = self.get(store_name)?;
 
-        let store_key = store.model.model_ndarray(&store_input);
-        let metadata_value: MetadataValue = store_input.into();
+        let store_input_type: AIStoreInputTypes = store_input.clone().into();
```

Comment:
No need to clone the entire store input just to turn it into an `AIStoreInputType` , especially because these inputs can be large (i.e strings and image buffers), you can instead have a `From<&StoreInput> for AIStoreInputType`